client/allocrunner/taskrunner/envoybootstrap_hook.go

```diff
@@ -148,3 +156,14 @@ func (h *envoyBootstrapHook) Prestart(ctx context.Context, req *interfaces.TaskP
 	resp.Done = true
 	return nil
 }
+
+func buildEnvoyAdminBind(alloc *structs.Allocation, taskName string) string {
+	port := envoyBaseAdminPort
+	for idx, task := range alloc.Job.LookupTaskGroup(alloc.TaskGroup).Tasks {
+		if task.Name == taskName {
+			port += idx
+			break
+		}
+	}
+	return fmt.Sprintf("localhost:%d", port)
+}
```

Comment:
If we do this in the job admission controller we can set the port as a group meta variable and interpolate it into the command. This would allow @jippi to interpolate it into the environment (or templates or command args) of other tasks in the same group.

We could also validate the port doesn't conflict with the port specified on the connect service (or `local_service_port`) or `local_bind_ports` for upstreams.

That's an awful lot more work, so we could write a test, get this merged, and then migrate to the admission controller in 0.11.x or something.

Reply:
I went ahead and injected an env var `NOMAD_ENVOY_ADMIN_ADDR_<service>` in from the hook env, let me know what you think. I don't love the idea of polluting the alloc environment with another variable but this seems reasonable until we can rework it into the admission controller?